workflow/torch/requires_grad.py

```diff
@@ -9,20 +9,17 @@ def __init__(self, module, grad=True):
 
         self.module = module
         self.grad = grad
-        self.requires_grads = [
-            p.requires_grad for p in module.parameters()
-        ]
+        self.named_requires_grads = { name: parameter.requires_grad for (name, parameter) in module.named_parameters() }
 
     def __enter__(self):
-        for p in self.module.parameters():
-            p.requires_grad = self.grad
+        for (_, parameter) in self.module.named_parameters():
```

Comment:
Any reason to use `named_parameters` here?

The tuple with brackets is also contrary to our general style. Do you want to discuss the benefits of this?

Reply:
sorry, I do like that naturally, probably from my years with Rust x)

Reply:
personally I think that is more consistent since the tuple unpacking is explicit and doesn't require a new syntax, but that's such a minor thing I don't even care